simplecpp.cpp

```diff
@@ -1282,14 +1270,74 @@ void simplecpp::TokenList::constFoldBitwise(Token *tok)
                 result = (stringToLL(tok->previous->str()) ^ stringToLL(tok->next->str()));
             else /*if (*op == '|')*/
                 result = (stringToLL(tok->previous->str()) | stringToLL(tok->next->str()));
-            tok = tok->previous;
-            tok->setstr(toString(result));
-            deleteToken(tok->next);
-            deleteToken(tok->next);
+            simpleSquash(tok, toString(result));
         }
     }
 }
 
+void simplecpp::TokenList::simpleSquash(Token *&tok, const std::string & result)
+{
+    tok = tok->previous;
+    tok->setstr(result);
+    deleteToken(tok->next);
+    deleteToken(tok->next);
+}
+
+void simplecpp::TokenList::squashTokens(Token *&tok, const std::set<std::string> & breakPoints, bool forwardDirection, const std::string & result)
+{
+    const char * const brackets = forwardDirection ? "()" : ")(";
+    Token* Token::* const step = forwardDirection ? &Token::next : &Token::previous;
+    int skip = 0;
+    const Token * const tok1 = tok->*step;
+    while (tok1 && tok1->*step) {
+        if ((tok1->*step)->op == brackets[1]){
+            if (skip) {
+                --skip;
+                deleteToken(tok1->*step);
+            } else
+                break;
+        } else if (skip) {
```

Comment:
so if skip is true here then we don't count inner brackets that seems unfortunate